File: 287-find-the-duplicate-number/find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        int ans;
        vector<int>freq(n+1,0);
        for(int it:nums){
            freq[it]++;
        }
        for(int i=0;i<=n;i++){
            if(freq[i]>1){
                ans=i;
                break;
            }
        }
        return ans;
    }
};

File: 287-find-the-duplicate-number/find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n= nums.length;
        int[] freq = new int[n+1];
        for(int it:nums){
            if(freq[it]==0){
                freq[it]++;
            }
            else return it;
        }
        return 0;
    }
}